moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -459,51 +459,55 @@ KinematicState Servo::getNextJointState(const ServoInput& command)
       target_state = haltJoints(joints_to_halt, current_state, target_state);
     }
   }
+  else
+  {
+    robot_state->copyJointGroupPositions(joint_model_group, target_state.positions);
```

Comment:
I'm not sure I see the bug here. L397 already does pretty much the same thing, right? Is `robot_state` modified after L397? I don't think so, jointDeltaFromCommand() doesn't change it.

Also, what about velocities like at L398?

Also, can the `robot_state` parameter of jointDeltaFromCommand() be `const`? That would eliminate the confusion.

Reply:
Good questions

* L397 copies the `robot_state` into the `current_state` variable (for computing the joint deltas), wheras this line copies the `robot_state` into the `target_state` variable (for output). Without this line, the case of halting for collision would leave `target_state` at all zeros.
* I thought about that, but don't we want the target state velocities to be all zeros in the case of a hard stop?

Reply:
Actually, for your first question, you're right. We do not need to have this else statement and instead we just don't publish commands in this status.